Sources/BraintreeCore/BTHTTP.swift

```diff
@@ -159,6 +154,50 @@ class BTHTTP: NSObject, URLSessionTaskDelegate {
             return
         }
     }
+    
+    
+    func httpRequest(
+        method: String,
+        path: String,
+        configuration: BTConfiguration? = nil,
+        parameters: Encodable? = nil,
+        headers: [String: String]? = nil
+    ) async throws -> (BTJSON?, HTTPURLResponse?) {
+        let request = try createRequest(
+            method: method,
+            path: path,
+            configuration: configuration,
+            parameters: parameters,
+            headers: headers
+        )
+        let (data, response) = try await session.data(for: request)
+        guard let httpResponse = createHTTPResponse(response: response) else {
+            throw BTHTTPError.httpResponseInvalid
+        }
+        guard let data = data as Data? else {
```

Comment:
I think this will always pass since the `as Data?` cast will always succeed since data is already Data type from `session.data(for:)`, this check would never fail. If the intent is to verify data exists, the current code doesn't do that since `Data()` is a valid empty data object. Are you able to take a look?